util/argo/argo.go

```diff
@@ -114,6 +115,57 @@ func FilterByProjects(apps []argoappv1.Application, projects []string) []argoapp
 	return items
 }
 
+// FilterByFile returns applications affected by files
+func FilterByFile(apps []argoappv1.Application, file []string) []argoappv1.Application {
+	// if no files  are provided, return all the apps
+	if len(file) == 0 {
+		return apps
+	}
+	items := make([]argoappv1.Application, 0)
+	existing := make([]bool, len(apps))
+	for i := 0; i < len(file); i++ {
+		filePath := filepath.Clean(file[i])
+		for j := 0; j < len(apps); j++ {
+			a := apps[j]
+			appPath := filepath.Clean(a.Spec.GetSource().Path)
+			annotations := strings.Split(a.GetAnnotation(argoappv1.AnnotationKeyManifestGeneratePaths), ";")
+			for _, annotation := range annotations {
+				annotation = strings.TrimSpace(annotation)
+				if annotation == "" {
+					// TODO: Call FilterByPath to match identical file and path args.
+					continue
+				}
+				var targetPath string
+
+				switch annotation {
+				case ".":
+					targetPath = appPath
+				case "..":
+					targetPath = filepath.Dir(appPath)
+
+				default:
+					if filepath.IsAbs(annotation) {
+						targetPath = filepath.Clean(annotation)
+					} else {
+						targetPath = filepath.Join(appPath, annotation)
+						targetPath = filepath.Clean(targetPath)
+					}
+				}
```

Comment:
ideally the default switch case should include no if blocks and other conditions. 
it should just return the default value of a swith case


```suggestion
				switch {
				case annotation == ".":
					targetPath = appPath
				case annotation == "..":
					targetPath = filepath.Dir(appPath)
				case filepath.IsAbs(annotation):
					targetPath = filepath.Clean(annotation)

				default:
						targetPath = filepath.Join(appPath, annotation)
						targetPath = filepath.Clean(targetPath)
				}
```

Reply:
@ppapapetrou76 Thank you for this tip!
I applied it.